qemu/tests/driver_in_use.py

```diff
@@ -0,0 +1,155 @@
+            if bool(check_reg.findall(output)):
+                return True
+        except Exception, err:
+            raise exceptions.TestFail("Check process error: %s" % err)
+        return False
+
+    def bg_stress_test():
+        """
+        Run backgroud test.
+
+        :return: return the background case thread if it's successful;
+                 else raise error.
+        """
+        bg_stress_test = params.get("run_bgstress")
+        if bg_stress_test:
+            try:
+                error_context.context("Run test %s background" % bg_stress_test,
+                                      logging.info)
+                stress_thread = ""
+                wait_time = float(params.get("wait_bg_time", 60))
+                target_process = params.get("target_process", "")
+                bg_stress_run_flag = params.get("bg_stress_run_flag")
+                env[bg_stress_run_flag] = False
+                stress_thread = utils.InterruptedThread(
+                    utils_test.run_virt_sub_test, (test, params, env),
+                    {"sub_type": bg_stress_test})
+                stress_thread.start()
+                if stress_thread.is_alive():
+                    try:
+                        stop_time = time.time() + wait_time
+                        while time.time() < stop_time:
+                            status = check_bg_running(session, target_process)
+                            if status:
+                                env["bg_status"] = 1
+                                break
+                    except Exception:
+                        raise exceptions.TestFail("Backgroud test %s is not"
+                                                  "alive!" % bg_stress_test)
+                if not utils_misc.wait_for(lambda: env.get(bg_stress_run_flag),
+                                           wait_time, 0, 5,
+                                           "Wait %s test start" % bg_stress_test):
+                    err = "Fail to start %s test" % bg_stress_test
+                    raise exceptions.TestError(err)
+                return stress_thread
+            except Exception:
+                raise exceptions.TestFail("Background case %s start failed!" %
+                                          bg_stress_test)
+
+    def run_subtest(run_bg_flag):
+        """
+        Run sub test. e.g. reboot / system_reset...
+
+        :params: run_bg_flag: a flag specified when to run the background case.
+        """
+        sub_type = params.get("sub_test")
+        if sub_type:
+            try:
+                msg = "Run sub test %s %s" % (sub_type, run_bg_flag)
+                error_context.context(msg, logging.info)
+                utils_test.run_virt_sub_test(test, params, env, sub_type)
+            except Exception:
+                raise exceptions.TestFail("Test failed when run %s!" % sub_type)
+
+    driver = params.get("driver_name")
```

Comment:
it looks driver_name is must, so I think params["driver_name"] is necessary here.